tests/test_edf.py

```diff
@@ -1238,10 +1248,142 @@ def test_update_record_duration_raises_error_if_signal_duration_is_not_exactly_d
         edf.update_data_record_duration(0.3)
 
 
-def test_update_record_duration_raises_error_if_sampling_rate_incompatible():
+def test_update_data_record_duration_raises_error_if_sampling_rate_incompatible():
     edf = Edf([EdfSignal(np.arange(100), 10), EdfSignal(np.arange(100), 10)])
     with pytest.raises(
         ValueError,
         match="Cannot set data record duration to 0.05: Incompatible sampling frequency 10.* Hz",
     ):
         edf.update_data_record_duration(0.05)
+
+
+@pytest.mark.parametrize("record_duration", [0, -1])
+def test_update_data_record_duration_raises_error_if_data_record_duration_not_positive(
+    record_duration: float,
+):
+    edf = Edf([EdfSignal(np.arange(100), 10)])
+    with pytest.raises(ValueError, match="Data record duration must be positive"):
+        edf.update_data_record_duration(record_duration)
+
+
+def test_update_data_record_duration_with_annotations(tmp_path: Path):
+    edf_file = tmp_path / "test.edf"
```

Comment:
We could use our custom fixture `tmp_file` instead of `tmp_path` to avoid this line